x/earn/keeper/withdraw.go

```diff
@@ -150,3 +151,175 @@ func (k *Keeper) Withdraw(
 
 	return nil
 }
+
+// WithdrawFromModuleAccount removes the amount of supplied tokens from a vault and transfers it
+// back to the module account.
+func (k *Keeper) WithdrawFromModuleAccount(ctx sdk.Context, from sdk.AccAddress, wantAmount sdk.Coin, withdrawStrategy types.StrategyType) error {
```

Comment:
Any chance this method could be refactored to avoid duplication with normal Withdraw? Otherwise its quite a lot of code without tests.

Reply:
Maybe? Worth noting it gets hit in the end-to-end test in proposal_handler_test.go, but we could also add unit tests